spec/worker_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+$result =~ nil
```

Comment:
**suggestion (non-blocking):** As I've promised, here's how one can use [testing double](https://relishapp.com/rspec/rspec-mocks/v/3-11/docs/basics/test-doubles) to assert a certain method has been called :sweat_smile: 

```ruby
it "should call #perform from child instance when #process_message is called" do
  broker = instance_double("MockBroker")
  allow(TestWorker).to receive(:build_instance_broker).and_return(broker)

  TestWorker.process_message
  expect(broker).to have_received(:poller)
end
```

Reply:
Really like this stubbing approach 😄